price/src/price_generator.rs

```diff
@@ -220,19 +219,24 @@ impl PriceGenerator {
 pub async fn get_price(
     client: &RpcClient,
     price_key: &SolPubkey,
-    age: u64,
     token_type: BlockchainTokenTypeV1,
 ) -> Result<Price> {
-    let mut price_account = client.get_account(price_key)?;
-    tracing::debug!("price_account: {:?}", price_account);
+    let price_oracle_v0_data = client.get_account_data(price_key).await?;
+    let mut price_oracle_v0_data = price_oracle_v0_data.as_ref();
+    let price_oracle_v0 = PriceOracleV0::try_deserialize(&mut price_oracle_v0_data)?;
 
     let current_time = Utc::now();
     let current_timestamp = current_time.timestamp();
-    let price_feed = load_price_feed_from_account(price_key, &mut price_account)?;
-    tracing::debug!("price_feed: {:?}", price_feed);
 
-    price_feed
-        .get_price_no_older_than(current_timestamp, age)
-        .map(|feed_price| Price::new(current_time, feed_price.price as u64, token_type))
-        .ok_or_else(|| anyhow!("unable to fetch price"))
+    let price = calculate_current_price(&price_oracle_v0.oracles, current_timestamp);
+
+    match price {
+        Some(p) => {
+            tracing::debug!("got price: {:?}", p);
+            Ok(Price::new(current_time, p, token_type))
+        }
+        None => {
+            bail!("unable to fetch price!")
+        }
+    }
```

Comment:
You could remove this match in favor of this if you want to
```rust
    calculate_current_price(&price_oracle_v0.oracles, current_timestamp)
        .map(|p| {
            tracing::debug!("got price: {:?}", p);
            Price::new(current_time, p, token_type)
        })
        .ok_or_else(|| anyhow!("unable to fetch price"))
```

Reply:
updated [8b12027](https://github.com/helium/oracles/pull/432/commits/8b12027334f64411dc64d4d18caeb026dcd6c795)